packages/flutter_tools/lib/src/device.dart

```diff
@@ -490,6 +491,33 @@ abstract class Device {
     await descriptions(devices).forEach(globals.printStatus);
   }
 
+  static Future<void> printDevicesAsJson(List<Device> devices) async {
+    globals.printStatus(
```

Comment:
I would move this logic into the devices command since that is the only place it is used

Reply:
I was actually confused about why `printDevices()` is a static method of the `Device`, so I decided to put mine alongside just to blend in :-)